src/eureka/S6_planet_spectra/s6_spectra.py

```diff
@@ -801,20 +907,103 @@ def compute_fn(meta, log, fit_methods, nsamp=1e4):
+        if ampcos.shape[-1] == 0:
+            # The parameter could not be found - skip it
+            log.writelog(f'  Parameter {meta.y_param} was not in the list of '
+                         'fitted parameters')
+            log.writelog(f'  Skipping {y_param}')
+            return meta
+
+    # Reset meta.y_param
+    meta.y_param = y_param
+
+    meta.spectrum_median = []
+    meta.spectrum_err = []
+
+    for i in range(meta.nspecchan):
+        fluxes = fp[i]*(1-2*ampcos[i])
+        flux = np.percentile(np.array(fluxes), [16, 50, 84])[[1, 2, 0]]
+        flux[1] -= flux[0]
+        flux[2] = flux[0]-flux[2]
+        meta.spectrum_median.append(flux[0])
+        meta.spectrum_err.append(flux[1:])
+
+    # Convert the lists to an array
+    meta.spectrum_median = np.array(meta.spectrum_median)
+    if meta.fitter == 'lsq':
+        meta.spectrum_err = np.ones((2, meta.nspecchan))*np.nan
+    else:
+        meta.spectrum_err = np.array(meta.spectrum_err).T
+
+    return meta
+
+
+def compute_fn_starry(meta, log, fit_methods, nsamp=1e3):
+    nsamp = int(nsamp)
+
+    # Save meta.y_param
+    y_param = meta.y_param
+
+    # Load eclipse depth
+    meta.y_param = 'fp'
+    fp = load_s5_saves(meta, log, fit_methods)
+    if fp.shape[-1] == 0:
         # The parameter could not be found - skip it
         log.writelog(f'  Parameter {meta.y_param} was not in the list of '
                      'fitted parameters')
         log.writelog(f'  Skipping {y_param}')
         return meta
 
+    nsamp = min([nsamp, len(fp[0])])
+    inds = np.random.randint(0, len(fp[0]), nsamp)
+
+    class temp_class:
+        def __init__(self):
+            pass
+
+    # Load map parameters
+    if not hasattr(meta, 'ydeg'):
+        meta.ydeg = 2  # For backwards compatibility with my old saves
+    temp = temp_class()
+    for ell in range(1, meta.ydeg+1):
+        for m in range(-ell, ell+1):
+            meta.y_param = f'Y{ell}{m}'
+            val = load_s5_saves(meta, log, fit_methods)
+            if val.shape[-1] != 0:
+                setattr(temp, f'Y{ell}{m}', val[:, inds])
+
     # Reset meta.y_param
     meta.y_param = y_param
 
+    # If no parameters could not be found - skip it
+    if len(temp.__dict__.keys()) == 0:
+        log.writelog('  No Ylm parameters were found...')
+        log.writelog(f'  Skipping {y_param}')
+        return meta
+
     meta.spectrum_median = []
     meta.spectrum_err = []
 
+    planet_map = starry.Map(ydeg=meta.ydeg, nw=nsamp)
+    planet_map2 = starry.Map(ydeg=meta.ydeg, nw=nsamp)
     for i in range(meta.nspecchan):
-        fluxes = fp[i]*(1-2*ampcos[i])
+        inds = np.random.randint(0, len(fp[i]), nsamp)
+        for ell in range(1, meta.ydeg+1):
+            for m in range(-ell, ell+1):
+                if hasattr(temp, f'Y{ell}{m}'):
+                    planet_map[ell, m, :] = getattr(temp, f'Y{ell}{m}')[i]
+                    planet_map2[ell, m, :] = getattr(temp, f'Y{ell}{m}')[i]
```

Comment:
I don't see the need for planet_map2, but I trust you've thought this through.

Reply:
This snippet is copied from Stage 5, so I'll explain the motivation for writing it that way for Stage 5. Because the code gets compiled (aka this for-loop isn't run at every step in the fit - only once before the fit is run), I was worried about circular references with setting the map's amplitude to a value that depends on it's amplitude. Specifically, `planet_map.flux()` depends on the value of `planet_map.amp`, so setting `planet_map.amp` based on `planet_map.flux()` with this compiled theano code had me worried about what would happen in practice as it would likely depend on the order in which the variables are updated. I just tried removing `planet_map2` though and it doesn't seem to fundamentally break things, but comparing two tests with `planet_map2` and two tests without it, the fit without `planet_map2` seemed to have a much harder time converging (many more divergences and large discrepancies between independent chains) and took longer to complete. Since there's some intrinsic variability in the path taken from one run to the next, it's maybe possible that the fit without `planet_map2` just followed an especially unlucky path twice, but I think that unlikely and either way it seems clear that there's no significant disadvantage to leaving things the way they currently are